src/build_targets.rs

```diff
@@ -66,6 +66,24 @@ impl BuildTargets {
         libkinds: &[&str],
         capi_config: &CApiConfig,
         use_meson_naming_convention: bool,
+    ) -> anyhow::Result<BuildTargets> {
+        Self::new_help(
+            name,
+            target,
+            targetdir,
+            LibraryTypes::from_library_types(target, libkinds.iter()),
+            capi_config,
+            use_meson_naming_convention,
+        )
+    }
```

Comment:
this duplication is to prevent changes to the public api (which would require major version bumps). If that is not a concern then that would save a bunch of lines here. Of course this can always be resolved at some point in the future.

Reply:
It is fairly unlikely that somebody uses cargo-c as library, I'd call it `new_inner` while at it.

Reply:
well if you don't care about bumping the major version then this intermediate layer can just be removed right?

Reply:
it isn't a concern IMHO, if somebody complains because it is using cargo-c as library I'll bump the version. The target users shouldn't have problems anyway.